### ISSUE TITLE
bro, you broke our build

### ISSUE THREAD

**Issue by dj8yfo** [DaniPopes/const-hex#17]
linking to https://github.com/nearuaguild/abstract-dao/pull/9 
and https://github.com/rust-lang/rust/issues/133290 , 

i'm still not convinced this is not a compiler problem, but as per spec here https://rust-lang.github.io/rfcs/2045-target-feature.html , `#[target_feature(enable = "simd128")]` is only allowed on `unsafe` functions , not sure how up to date that spec is

https://github.com/DaniPopes/const-hex/blob/85f08142ccc5300c90faa94484b20db19388bf19/src/arch/wasm32.rs#L71-L72

![image](https://github.com/user-attachments/assets/c51fb8a6-7047-45da-8922-ff464718caa0)

and `portable-simd` feature wasn't called for in this use case https://docs.rs/crate/ethers-contract/2.0.14/source/Cargo.toml#24

```bash
❯ : cargo tree -i const-hex
const-hex v1.13.1
├── ethers-contract v2.0.14
│   └── near_abstract_dao v0.6.1 (/home/user/Documents/code/abstract-dao)
└── ethers-core v2.0.14
    ├── ethers-contract v2.0.14 (*)
    └── near_abstract_dao v0.6.1 (/home/user/Documents/code/abstract-dao)
```

**Comment by hanna-kruppe** [comment]
More constructively: I see that *calls* to the functions using simd128 are gated on `if has_simd128()`, which is defined as:

https://github.com/DaniPopes/const-hex/blob/85f08142ccc5300c90faa94484b20db19388bf19/src/arch/wasm32.rs#L9-L12

That's the right spirit, but unfortunately, it's not guaranteed that this style of `cfg`-gating actually omits the functions in question when simd128 is not available. See also https://github.com/BurntSushi/memchr/issues/144 for a similar problem reported before. In general it's very likely to work most of the time, but evidently not always, especially (but not only) with `opt-level = 0`. The only reliable way to ensure that the wasm module doesn't contain any simd128 instructions, and thus can be loaded by engines that don't implement SIMD, is to put `#[cfg(target_feature = "simd128")]` on those functions. Everything else depends on compiler flags and undocumented implementation details of the compiler (e.g., how smart it tries to be about "which functions need to get codegen'd").

I don't know why `check_simd128` can be defined without `unsafe` but I don't think it matters.

**Comment by DaniPopes** [comment]
Thanks for the report. That's unfortunate considering it is the same implementation as the other architectures. I'll apply the same fix as in BurntSushi/memchr#144.

As for "without `unsafe`", that's because most of the intrinsics can't cause undefined behavior so they're not marked `unsafe`, however I can't find where this is explained.

Also I would suggest a more descriptive issue name for next time :D

**Comment by DaniPopes** [comment]
Released in 1.13.2.
